[PG NCCL][RFC] Dont crash on retrieveDesyncReport failure

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #97702
* #97692
* __->__ #97548
* #97544
* #97462

Failing to generate this report should not crash.

Differential Revision: [D44376707](https://our.internmc.facebook.com/intern/diff/D44376707/)

rohan-varma committed:
[PG NCCL][RFC] Dont crash on retrieveDesyncReport failure

Failing to generate this report should not crash.

Differential Revision: [D44376707](https://our.internmc.facebook.com/intern/diff/D44376707/)

[ghstack-poisoned]

kwen2501 commented:
Mitigates https://github.com/pytorch/pytorch/issues/97268

kwen2501 commented on torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp:
```diff
@@ -831,7 +831,11 @@ void ProcessGroupNCCL::abortTimedOutCollectives(
+            exceptionMsg += retrieveDesyncReport(store_, "NCCL", rank_, size_);
+        catch (const std::exception &e) {
+            LOG(ERROR) << "Failed to retrieve NCCL desync report. Please report an issue to PyTorch Distributed.";
+        }
```
nit: maybe print `e.what()` to get more information?

rohan-varma committed:
Update on "[PG NCCL][RFC] Dont crash on retrieveDesyncReport failure"


Failing to generate this report should not crash.

Differential Revision: [D44376707](https://our.internmc.facebook.com/intern/diff/D44376707/)

[ghstack-poisoned]

kwen2501 committed:
Lint
